tests/baselines/reference/spreadObjectOrFalsy.js

```diff
@@ -111,7 +111,7 @@ declare function f5<S, T extends undefined>(a: S | T): S | T;
 declare function f6<T extends object | undefined>(a: T): T;
 declare function g1<T extends {}, A extends {
     z: (T | undefined) & T;
-}>(a: A): (T | undefined) & T;
+}>(a: A): T | (undefined & T);

```

Comment:
Maybe an aside, but a display question: We explicitly choose to list `undefined` at the end of a union, rather than the beginning as its type id would usually imply. Should we do the same for intersections? Looks kinda odd to see `undefined & T` instead of `T & undefined`.

Reply:
Should we do the same for intersections? Probably not, given that intersections are ordered.

Reply:
But they're ordered because different overload merge orderings is meaningful - I don't think differing orders of intersecting `undefined` are ever meaningful.